perfkitbenchmarker/providers/docker/docker_container_instance.py

```diff
@@ -0,0 +1,319 @@
+      self.RemoteCommand(https_proxy % FLAGS.http_proxy)
+    if FLAGS.ftp_proxy:
+      ftp_proxy = "sed -i '1i export ftp_proxy=%s' /etc/bash.bashrc"
+      self.RemoteCommand(ftp_proxy % FLAGS.ftp_proxy)
+
+  def _BuildVolumesBody(self):
+    """Constructs volumes-related part of create command for Docker Container"""
+    volumes = []
+
+    for scratch_disk in self.scratch_disks:
+      vol_string = scratch_disk.volume_name + ":" + scratch_disk.mount_point
+      volumes.append('-v')
+      volumes.append(vol_string)
+
+    return volumes
+
+  def _LocalImageExists(self, docker_image_name):
+    """Returns whether a Docker image exists locally."""
+    inspect_cmd = ['docker', 'image', 'inspect', docker_image_name]
+    info, _, returnCode = vm_util.IssueCommand(inspect_cmd, suppress_warning=True)
+    info = json.loads(info)
+    logging.info("Checking if Docker Image Exists")
+    if len(info) > 0 and returnCode == 0:
+      logging.info("Image exists")
+      return True
+    logging.info("Image does not exist")
+    return False
+
+  def _BuildImageLocally(self):
+    """Build Container Image Locally
+
+    Dockerfiles located at PerfKitBenchmarker/data/docker/pkb/<containerImage>/Dockerfile
+    """
+    directory = os.path.dirname(
+        data.ResourcePath(os.path.join(DOCKERFILE_DIRECTORY,
+                                       self.container_image,
+                                       'Dockerfile')))
+    build_cmd = [
+        'docker', 'build', '--no-cache',
+        '-t', self.container_image, directory]
+
+    vm_util.IssueCommand(build_cmd)
+
+  def GetResourceMetadata(self):
+    """Returns a dict containing metadata about the VM.
+
+    Returns:
+      dict mapping string property key to value.
+    """
+    result = super(DockerContainer, self).GetResourceMetadata()
+    logging.warn("GET RESOURCE METADATA")
+    print(result)
+    return result
+
+
+class DebianBasedDockerContainer(DockerContainer,
+                                 linux_virtual_machine.DebianMixin):
+  DEFAULT_IMAGE = UBUNTU_IMAGE
+
+  def _GetNumCpus(self):
+    return self.cpus
+
+  def ApplySysctlPersistent(self, sysctl_params):
+    """Override ApplySysctlPeristent function for Docker provider
+
+    Parent function causes errors with Docker because it shutdowns container
+    Args:
+      sysctl_params: dict - the keys and values to write
+    """
+    logging.warn("sysctl flags are applied when container is created. "
+                 "Not all sysctl flags work with Docker. It does not "
+                 "support flags that modify the host system")
+
+  def _RebootIfNecessary(self):
+    """Override RebootIfNecessary for Docker Provider"""
+    logging.warn("Docker Containers cannot be rebooted to apply flags")
+
+
+class Ubuntu1604BasedDockerContainer(
+    DebianBasedDockerContainer, linux_virtual_machine.Ubuntu1604Mixin):
+  DEFAULT_IMAGE = UBUNTU_IMAGE
+
+# class Ubuntu1404BasedDockerVirtualMachine(
```

Comment:
Suggest dropping this commented out code and perhaps adding a note about what it will take to add support for Ubuntu 14 and Ubuntu 18.

Drop Ubuntu 17.10 since it is not a long-term support version and is no longer supported.